packages/rpc-methods/src/restricted/invokeSnap.ts

```diff
@@ -86,8 +85,8 @@ function getInvokeSnapImplementation({
     const snapIdString = method.substr(SNAP_PREFIX.length);
 
     if (!getSnap(snapIdString)) {
-      await addSnap({
-        id: snapIdString,
+      throw ethErrors.rpc.invalidRequest({
+        message: `The snap "${snapIdString}" is not installed. This is a bug, please report it.`,
```

Comment:
Edited this message to solicit a bug report. We're in a bad state if something has the permission to talk to a snap that we don't have.